Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -696,9 +731,8 @@ noncomputable def add : W.Point → W.Point → W.Point
   | 0, P => P
   | P, 0 => P
   | @some _ _ _ x₁ y₁ h₁, @some _ _ _ x₂ y₂ h₂ =>
-    if hx : x₁ = x₂ then
-      if hy : y₁ = W.negY x₂ y₂ then 0 else some <| nonsingular_add h₁ h₂ fun _ => hy
-    else some <| nonsingular_add h₁ h₂ fun h => (hx h).elim
+    if h : x₁ = x₂ ∧ y₁ = W.negY x₂ y₂ then 0
```

Comment:
You beat me to it! [Here is my incomplete branch](https://github.com/leanprover-community/mathlib4/compare/EllipticCurve.Chore) that does some of the golfing for affine point addition, but I haven't made a PR because I felt they were still somewhat inconsequential. Maybe you can have a look and get something out of it, and I'll update my branch with more chores in the future once this one is merged.

Reply:
In this PR I'm just golfing the definition and fixing things, so I basically only applied forced golfs, having no intention to make the diff larger. Once this PR is accepted, your further golf of the lemmas should be easy to merge!